core/src/parser.rs

```diff
@@ -202,6 +202,16 @@ fn parse_struct(s: &ItemStruct) -> Result<RustItem, ParseError> {
             if f.unnamed.len() > 1 {
                 return Err(ParseError::ComplexTupleStruct);
             }
+
+            if f.unnamed.is_empty() {
+                return Ok(RustItem::Alias(RustTypeAlias {
```

Comment:
Shouldn't this be basically the same as a Unit struct?

Reply:
It is, I just needed an early return ;). If you have a suggestion how to move things around to remove the duplication I'm open for changes!

Reply:
Sorry, I more mean that Unit structs are `RustItem::Struct`, but this is using a `RustItem::Alias`

Reply:
I can change that, but then the representation of:
```
#[derive(Debug, serde::Serialize)]
#[typeshare::typeshare]
pub struct MyType();
```

and 

```
#[derive(Debug, serde::Serialize)]
#[typeshare::typeshare]
pub struct MyType(pub ());
```

would differ.

Currently
```
#[derive(Debug, serde::Serialize)]
#[typeshare::typeshare]
pub struct MyType(pub ());
```

results in:

```
typealias MyType = Unit
```

in Kotlin. If I change it, it'll result in:

```
@Serializable
object MyType
```

In my opinion those representations should be the same, we can change either one of them. Which one would you prefer?  I added those two cases into the snapshot test in the PR.

Reply:
Actually, that's a great catch– we need to make sure that we're also consistent with the format that `serde` will use during serialization and deserialization. It looks like serde treats all tuple structs with length `0|2+` as arrays:

```
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ef7881f4313744aa43e5797a569ceb0c

Reply:
Actually, that's a great catch; we need to make sure that the chosen representation aligns with how serde will treat the type. It appears that serde treats empty tuple structs the same was as non-newtype tuple structs:

```rust
#[derive(serde::Serialize)]
struct Unit;

#[derive(serde::Serialize)]
struct Tup0();

#[derive(serde::Serialize)]
struct Tup1(i32);

#[derive(serde::Serialize)]
struct Tup2(i32, i32);

fn main() {
    println!("{}", serde_json::to_string(&Unit).unwrap());
    println!("{}", serde_json::to_string(&Tup0()).unwrap());
    println!("{}", serde_json::to_string(&Tup1(3)).unwrap());
    println!("{}", serde_json::to_string(&Tup2(4, 5)).unwrap());
}
```

```
null
[]
3
[4,5]
```

[playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ef7881f4313744aa43e5797a569ceb0c)

There's probably not a useful way to handle this round trip through an array in the destination language (for the same reason we don't support complex tuple structs), so we have to close this for now. Any support for empty tuple structs would also have to include support for larger, 2+ tuple structs, and some brief experimenting revealed that this would be very challenging in some languages (Swift especially).

Thanks for working through this with us. Discoveries like this are valuable for ensuring we don't back ourselves into a corner by supporting rust type shapes without clear analogues in other languages.

Reply:
Thanks for the info. If you need any further help with this issue feel free to contact me.